### ISSUE TITLE
Don't have support for resize(H5Dset_extent)

### ISSUE THREAD

**Issue by zyc-sudo** [gonum/hdf5#53]
Hi,
I tried to find the wrapper for H5Dset_extent which resize the chunked datasets. I forked the repo to https://github.com/zyc-sudo/hdf5 and implemented the wrapper in h5d_dataset.go as 
func (s *Dataset) Resize(dims []uint)

I'm not sure if you guys already implemented it or not. If you already did, could you please tell me where it is? Or if you didn't let me know too so we can do pull request when I finish it to contribute a little to this project

**Comment by kortschak** [comment]
We have not implemented that. However, the implementation you have is not currently safe. What happens if someone calls `s.Resize([]uint{})`?

**Comment by zyc-sudo** [comment]
Thank you for your response. You are correct, it is not safe at all and I’m
new to golang also. I was doing another project and need this feature so
badly that I created it in a “I just want it to work” mind set. I’ll
perfect that with proper error handling and input validation for sure.
Thank you very much for your response and I’ll keep you posted when I
actually have a better version of it. I hope I can contribute to this
project when I implement a better version.

**Comment by kortschak** [comment]
Yes, please do. I'll go through the code with comments with a view to merging this when I get some time.

**Comment by zyc-sudo** [comment]
Thanks!

**Comment by mrwill84** [comment]
wow good project . 
Bros, it's merged ?

**Comment by zyc-sudo** [comment]
> 

Nope. I think the project is stalled. If you want to use it, you can use my fork where I implemented it.

**Comment by zyc-sudo** [comment]
> wow good project .
> Bros, it's merged ?

Nope. I think the project is stalled. If you want to use it, you can use my fork where I implemented it.

**Comment by mrwill84** [comment]
@zyc-sudo thank you ;)

**Comment by mrwill84** [comment]
@zyc-sudo man, since your repo can't open issue to support,  could you take a quick look at   ? it failed when I want to resize dim . 
https://gist.github.com/mrwill84/66f47b5e4dc2e9de5fdc45de69983e09

**Comment by zyc-sudo** [comment]
It needs a “max size” when you create the dataspace instead of nil.
You put nil:
hdf5.CreateSimpleDataspace(dims, nil)
It should not be nil.
Have a look at this file:
https://github.com/zyc-sudo/hdf5/blob/master/h5d_dataset_test.go

There is a test code :
func TestDatasetResize
In this function you can see you need to specify the maximum size the dataset can be resized to:

https://github.com/zyc-sudo/hdf5/blob/036a2c6fedc9ec44f82aef381e8e6e1d3679391f/h5d_dataset_test.go#L225
Also the chunk size need to be set too:

https://github.com/zyc-sudo/hdf5/blob/036a2c6fedc9ec44f82aef381e8e6e1d3679391f/h5d_dataset_test.go#L236
I recall that only “chunked” dataset can be resized. But you can refer to the hdf5 C api to verify this. 
this test code has passed so I know if you follow it , it works

**Comment by mrwill84** [comment]
thank you man
